public/app/features/migrate-to-cloud/onprem/useNotifyOnSuccess.tsx

```diff
@@ -0,0 +1,60 @@
+import { t } from 'i18next';
+import { useRef, useEffect } from 'react';
+
+import { useAppNotification } from 'app/core/copy/appNotification';
+
+import { GetSnapshotResponseDto, SnapshotDto } from '../api';
+
+export function useNotifySuccessful(snapshot: GetSnapshotResponseDto | undefined) {
+  const previousStatusRef = useRef<SnapshotDto['status']>(undefined);
+  const notifyApp = useAppNotification();
+
+  useEffect(() => {
+    const status = snapshot?.status;
+    const didJustFinish =
+      previousStatusRef.current !== 'FINISHED' && previousStatusRef.current !== undefined && status === 'FINISHED';
+
+    previousStatusRef.current = status; // must be AFTER the check above
+
+    if (!didJustFinish) {
+      return;
+    }
+
+    if (snapshot) {
+      const title = t('migrate-to-cloud.onprem.success-title', 'Migration completed!');
+      const message = getTranslatedMessage(snapshot);
+
+      notifyApp.success(title, message);
+    }
+  }, [notifyApp, snapshot]);
+}
+
+function getTranslatedMessage(snapshot: GetSnapshotResponseDto) {
+  const types: string[] = [];
+
+  for (const [type, count] of Object.entries(snapshot.stats?.types ?? {})) {
+    if (count <= 0) {
+      continue;
+    }
+
+    // We don't have per-resource status counts, so there's no way to accurately pluralize these
+    // so we just don't :)
```

Comment:
are we ever going to have per-resource status counts? 🤔 

if not, i wonder whether it's just easier to keep it generic and say `Successfully migrated 237 resources to your Grafana Cloud instance`. that also means you could properly pluralize and wouldn't have to keep updating this translated string each time a new resource type gets added. up2u 😄

Reply:
ppffffft yeah... You're right.

So originally, the idea was to use this to reinforce that only supported resource types were migrated, and link out to docs about migrating others.

However, we cannot dispatch notifications with HTML/react links in them, so that never really came to fruition. So yeah, it's probably not worth including the resources now.

But im happy to leave this here for now and revisit this later 👍